src/socket/dhcpv4.rs

```diff
@@ -247,6 +251,18 @@ impl<'a> Socket<'a> {
         self.ignore_naks = ignore_naks;
     }
 
+    /// Set the server/client port
+    ///
+    /// Allows you to specify the ports used by DHCP.
+    /// This is meant to support esoteric usecases allowed by the dhclient program.
+    pub fn set_ports(&mut self, ports: (u16, u16)) {
```

Comment:
I'm not sure if this should be done as a single method or two separate ones `(set_server_port`, `set_client_port`). We could also provide a structure with meaningful field names instead of a tuple. What do you think?

Reply:
Single method is fine. 

I'd do separate args instead of a tuple though, so the argument names tells which is which: `fn set_ports(&mut self, server_port, client_port)`. Same for the `ports` field in the struct.

Reply:
Done!